doc/sign-package.md

```diff
@@ -159,3 +159,95 @@ If the host arguments are not allowed to be passed then any specified within the
 If any host environment variables are configured as available within the SGX enclave, then this command will pass them though. Enclave specific environment variables will be added once Mystikos transfers control to the enclave.
 
 You then can ship your application in the form of this single exectuable to your desired platform.
+
+#### Packaging with EXT2 
+
+##### Signing EXT2 image
+
+Unlike packaging with a CPIO archive, an EXT2 image is not part of the signed package.
+Signing the EXT2 image allows us to ensure that Mystikos will only
+load a file system that matches the root hash or public key that we specified.
+
+> Signed EXT2 images are only supported for the root file system.
+Any additional file system mounted through configuration (auto-mount) or through
+calling `mount()` in the application (explicit mount) will not have the
+signature verification.
+
+
+The following command creates a signed EXT2 image.
+
+```
+$ myst mkext2 --sign=private.pem appdir ext2image
+```
+
+The ``private.pem`` file is the private key.
+
+The following generates test keys.
+
+```
+$ openssl genrsa -out private.pem -3 3072
+$ openssl rsa -in private.pem -pubout -out public.pem
+```
+
+##### Packaging with EXT2 image
+
+The EXT2 image created above can be loaded by a Mystikos application that
+trusts the given signing authority. This can be created with the following command.
+
+**Using roothash**:
+
+```
+$ myst package-sgx --roothash=roothash private.pem config.json
+```
+
+To obtain the roothash for an EXT2 image, use:
+
+```
+$ myst fssig --roothash <ext2image>
+```
+
+**Using public key**:
+
+```
+$ myst package-sgx --pubkey=public.pem private.pem config.json
+```
+
+The procedure is similar to packaging a directory or a CPIO archive, except
+that the ``--roothash`` option is used to add a root hash to the signed
+application (where the option argument is a file containing the root
+hash in ASCII format). This option may be repeated. The signed applicaiton
+maintains a list of **trusted root hashes** and only allows EXT2 images with
+those root hashes to be mounted.
+
+The ``--pubkey`` option adds a public key to the signed application,
+where public.pem is the public key of the signing authority. This
+option may be repeated. The signed application maintains a list of **trusted
+public keys** and only allows EXT2 images signed by those signers to be mounted.
+
+The advantage to trusted public keys (over trusted root hashes) is that the
+signed application trusts any EXT2 image that has is signed by a given
```

Comment:
1. It's Mystikos, not the application, trying to establish the trust.
1.  The public key doesn't necessarily have to be associated and distributed with a certificate.

Reply:
1. I believe `signed application` refers to the package that is generated by `myst package`, which includes Mystikos binary. I will reword for clarity. 
2. Certificate reference removed.